src/main/drivers/dshot.h

```diff
@@ -57,25 +57,43 @@ typedef struct dshotTelemetryQuality_s {
 extern dshotTelemetryQuality_t dshotTelemetryQuality[MAX_SUPPORTED_MOTORS];
 #endif // USE_DSHOT_TELEMETRY_STATS
 
-#define DSHOT_NORMAL_TELEMETRY_MASK     (1 << DSHOT_TELEMETRY_TYPE_eRPM)
-#define DSHOT_EXTENDED_TELEMETRY_MASK   (~DSHOT_NORMAL_TELEMETRY_MASK)
+#define DSHOT_NORMAL_TELEMETRY_MASK                 (1 << DSHOT_TELEMETRY_TYPE_eRPM)
+#define DSHOT_EXTENDED_TELEMETRY_MASK               (~DSHOT_NORMAL_TELEMETRY_MASK)
+#define DSHOT_TELEMETRY_RANGE_MASK                  (0x0f00u)
+#define DSHOT_TELEMETRY_VALUE_MASK                  (0x00ffu)
+#define DSHOT_TELEMETRY_STATUS_ALERT_EVENT_MASK     (0x80u)
+#define DSHOT_TELEMETRY_STATUS_WARNING_EVENT_MASK   (0x40u)
+#define DSHOT_TELEMETRY_STATUS_ERROR_EVENT_MASK     (0x20u)
+#define DSHOT_TELEMETRY_STATUS_MAX_STRESS_LVL_MASK  (0x0Fu)
+#define DSHOT_MAX_STRESS_LVL_WARNING_THRESHOLD      (9u)
+#define DSHOT_TELEMETRY_ENABLE_RESPONSE             (DSHOT_TELEMETRY_RANGE_STATUS)
 
 typedef enum dshotTelemetryType_e {
-    DSHOT_TELEMETRY_TYPE_eRPM           = 0,
-    DSHOT_TELEMETRY_TYPE_TEMPERATURE    = 1,
-    DSHOT_TELEMETRY_TYPE_VOLTAGE        = 2,
-    DSHOT_TELEMETRY_TYPE_CURRENT        = 3,
-    DSHOT_TELEMETRY_TYPE_DEBUG1         = 4,
-    DSHOT_TELEMETRY_TYPE_DEBUG2         = 5,
-    DSHOT_TELEMETRY_TYPE_DEBUG3         = 6,
-    DSHOT_TELEMETRY_TYPE_STATE_EVENTS   = 7,
+    DSHOT_TELEMETRY_TYPE_eRPM,
+    DSHOT_TELEMETRY_TYPE_TEMPERATURE,
+    DSHOT_TELEMETRY_TYPE_VOLTAGE,
+    DSHOT_TELEMETRY_TYPE_CURRENT,
+    DSHOT_TELEMETRY_TYPE_DEBUG1,
+    DSHOT_TELEMETRY_TYPE_DEBUG2,
+    DSHOT_TELEMETRY_TYPE_STRESS_LEVEL,
+    DSHOT_TELEMETRY_TYPE_STATUS,
     DSHOT_TELEMETRY_TYPE_COUNT
 } dshotTelemetryType_t;
 
+typedef enum dshotTelemetryRange_e {
```

Comment:
(isn't it possible to encode this as dshotTelemetryType_e + shift? It may make the code slightly simpler, at least with current implementation. )

Reply:
Sure it would be possible, but I fail to understand how it would make the code simpler (apart from getting rid of this one enum)?

Reply:
now there is arbitrary mapping between range, telemetryType and debugType. But all have similar values in identical order. 
If order is (defined as) fixed, all values can be converted into others using simple arithmetic operations.
no switch/table would be necessary..

Reply:
it's just an idea and there are other things that need fixing in bf.

IMO dshotTelemetryRange is 3bit value, shifted 9 bits. Topmost mantissa bit can be handled separately.

> is order in enums defined as fixed? Can I assume that for example debugType will only be appended to, never re-ordered?

it is common for enums that are shared to outside - debugType must match configurator and support some versions difference.
but in this case, you can use first group member and add offset(possibly even when defining enum members, but that is not common in bf). and add comment that these 8 items must stay together.

Reply:
(also see https://github.com/stylesuxx/betaflight/pull/1 for some possible ideas)

Reply:
Thank you - totally missed your PR.

Went through it, much more elegant. Just added a question for clarification.